datumaro/plugins/synthia_format.py

```diff
@@ -0,0 +1,160 @@
+    else: # only copy defined colors
+        label_id = lambda label: label_categories.find(label)[0]
+        colormap = { label_id(name): (desc[0], desc[1], desc[2])
+            for name, desc in label_map.items() }
+    mask_categories = MaskCategories(colormap)
+    mask_categories.inverse_colormap # pylint: disable=pointless-statement
+    categories[AnnotationType.mask] = mask_categories
+    return categories
+
+def parse_label_map(path):
+    if not path:
+        return None
+
+    label_map = OrderedDict()
+    with open(path, 'r', encoding='utf-8') as f:
+        for line in f:
+            # skip empty and commented lines
+            line = line.strip()
+            if not line or line and line[0] == '#':
+                continue
+
+            # color, name
+            label_desc = line.strip().split()
+
+            if 2 < len(label_desc):
+                name = label_desc[3]
+                color = tuple([int(c) for c in label_desc[:-1]])
+            else:
+                name = label_desc[0]
+                color = None
+
+            if name in label_map:
+                raise ValueError("Label '%s' is already defined" % name)
+
+            label_map[name] = color
+    return label_map
+
+class SynthiaExtractor(SourceExtractor):
+    def __init__(self, path):
+        if not osp.isdir(path):
+            raise FileNotFoundError("Can't read dataset directory '%s'" % path)
+
+        super().__init__()
+
+        self._categories = self._load_categories(path)
+        self._items = list(self._load_items(path).values())
+
+    def _load_categories(self, path):
+        label_map = None
+        label_map_path = osp.join(path, 'labels.txt')
+        if osp.isfile(label_map_path):
+            label_map = parse_label_map(label_map_path)
+        else:
+            label_map = SynthiaLabelMap
+        self._labels = [label for label in label_map]
+        return make_categories(label_map)
+
+    def _load_items(self, root_dir):
+        image_dir = osp.join(root_dir, 'RGB')
+        if osp.isdir(image_dir):
+            images = {
+                osp.splitext(osp.relpath(p, image_dir))[0].replace('\\', '/'): p
+                for p in find_images(image_dir, recursive=True)
+            }
+        else:
+            images = {}
+
+        items = {}
+
+        gt_dir = osp.join(root_dir, SynthiaPath.LABELS_SEGM_DIR)
+        if osp.isdir(gt_dir):
+            gt_images = find_images(gt_dir, recursive=True)
+            for gt_img in gt_images:
+                item_id = osp.splitext(osp.relpath(gt_img, gt_dir))[0].replace('\\', '/')
+
+                anno = []
+                instances_mask = load_image(gt_img, dtype=np.uint16)[:,:,2]
+                segm_ids = np.unique(instances_mask)
+                for segm_id in segm_ids:
+                    anno.append(Mask(
+                        image=self._lazy_extract_mask(instances_mask, segm_id),
+                        label=segm_id))
```

Comment:
This looks like this will merge together masks corresponding to different objects. Why not use the instance ID channel, which is mentioned in the description, to create separate masks for objects of different type?